File: Day_8/sol.java
class Solution {
    String reverseWords(String s) {
        StringBuilder ans = new StringBuilder();
        int n = s.length(), i=0;
        while(i<n){
            char c = s.charAt(i);
            
            if(c=='.'){
                i++;
                continue;
            }
            
            StringBuilder temp = new StringBuilder();
            while(i<n && s.charAt(i)!='.'){
                temp.append(s.charAt(i));
                i++;
            }
            
            ans.insert(0,temp+".");
        }
        
        return ans.toString().substring(0,n);
    }
}

File: Day_8/sol.js
/**
 * @param {string} str
 * @returns {string}
 */

class Solution {
    // Function to reverse words in a given string.
    reverseWords(str) {
        let ans = '';
        let n = str.length, i = 0;
        while (i < n) {
            let c = str.charAt(i);
            
            if (c === '.') {
                i++;
                continue;
            }
            
            let temp = '';
            while (i < n && str.charAt(i) !== '.') {
                temp += str.charAt(i);
                i++;
            }
            
            ans = temp + '.' + ans;
        }
        
        return ans.substring(0, n).replace(/\.$/, ''); // remove trailing dot
    }
}